internal/helper/sink_ack_ordering.go

```diff
@@ -0,0 +1,98 @@
+package helper
+
+import (
+	"context"
+
+	"github.com/uw-labs/substrate"
+	"golang.org/x/sync/errgroup"
+)
+
+var (
+	_ substrate.AsyncMessageSink = (*AckOrderingSink)(nil)
+)
+
+func NewAckOrderingSink(sink substrate.AsyncMessageSink) *AckOrderingSink {
+	return &AckOrderingSink{sink}
+}
+
+type AckOrderingSink struct {
+	innerSink substrate.AsyncMessageSink
+}
+
+func (s *AckOrderingSink) PublishMessages(ctx context.Context, acks chan<- substrate.Message, messages <-chan substrate.Message) error {
+	eg, ctx := errgroup.WithContext(ctx)
+
+	fromInner := make(chan substrate.Message, 1024)
```

Comment:
What's the reason for using buffered channels? And why is the capacity set to 1024?

Reply:
They don't _need_ to be buffered, but I (unsurprisingly) saw slightly better throughput with buffered due to fewer runtime context switches.  1024 is just a fairly randomly chosen Preferred Number.

Reply:
Fair enough, I guess it's fine on the producer side 👍